### PR TITLE
Refactor and adding more tests

### PR DESCRIPTION
**Pull request by philipktlin**
Refactor a bit and add more tests to increase code coverage.

### CODE REVIEW

File: tools/IoTEdgeDevOps/DevOpsLib/AgentMatrix.cs [hunk 1]
```diff
@@ -7,67 +7,60 @@ namespace DevOpsLib
...
         readonly IList<AgentDemandSet> rows = new List<AgentDemandSet>()
         {
             new AgentDemandSet(
                 "Linux AMD64",
                 new HashSet<AgentCapability>
                 {
-                    new AgentCapability(CapabilityKeys.AgentOS, "Linux"),
-                    new AgentCapability(CapabilityKeys.AgentOSArchitecture, "X64"),
+                    new AgentCapability(AgentCapabilityKeys.AgentOS, "Linux"),
+                    new AgentCapability(AgentCapabilityKeys.AgentOSArchitecture, "X64"),
```
Comment (yophilav) [inline tools/IoTEdgeDevOps/DevOpsLib/AgentMatrix.cs]: For `AgentOS`, please consider using `OSPlatform.Linux` 
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform?view=netframework-4.8

For `AgentOSArchitecture`, plase consider using `Architecture.X64`
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.architecture?view=netframework-4.8

File: tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeAgent.cs [hunk 2]
```diff
@@ -108,34 +108,13 @@ public bool Equals(IoTEdgeVstsAgent other)
...
-            }
-
-            return true;
+            return this.id == other.id &&
+                string.Equals(this.name, other.name, StringComparison.Ordinal) &&
+                string.Equals(this.version, other.version, StringComparison.Ordinal) &&
+                this.status == other.status &&
+                this.enabled == other.enabled &&
+                this.systemCapabilities.SetEquals(other.systemCapabilities) &&
+                this.userCapabilities.SetEquals(other.userCapabilities);
```
Comment (yophilav) [inline tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeAgent.cs]: Can this be replaced by comparing the hash from `GetHash()`?

File: tools/IoTEdgeDevOps/DevOpsLib/AgentMatrix.cs (hunk 1 above)
Comment (philipktlin) [reply tools/IoTEdgeDevOps/DevOpsLib/AgentMatrix.cs]: For OS platform, there is a value "Windows_NT" report from vsts agent, which can't match in System.Runtime.InteropServices.OSPlatform.  I will add TODO to see if we should create our own OS platform enum as we are going to support Yocto very soon.

File: tools/IoTEdgeDevOps/DevOpsLib/BuildDefinitionId.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace DevOpsLib
+{
+    public enum BuildDefinitionId
```
[the remark this answers is not in the document]
Comment (philipktlin) [reply tools/IoTEdgeDevOps/DevOpsLib/BuildDefinitionId.cs]: Go to each build definition from browser and build definition id will be shown as query parameter in URL.

File: tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeAgent.cs (hunk 2 above)
Comment (philipktlin) [reply tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeAgent.cs]: No.  Hash code of 2 instances can be the same but they may not be equal.

**Comment by yophilav** [comment]
Can you mention Dashboard in the PR title please?

**Pull request merged**